plugins/woocommerce/includes/class-wc-cart.php

```diff
@@ -1696,29 +1697,18 @@ public function check_customer_coupons( $posted ) {
 	 *
 	 * @param array $check_emails Array of customer email addresses.
 	 * @param array $restrictions Array of allowed email addresses.
+	 *
 	 * @return bool
+	 * @deprecated 9.0.0 In favor of static method Automattic\WooCommerce\Utilities\DiscountsUtil::is_coupon_emails_allowed.
 	 */
 	public function is_coupon_emails_allowed( $check_emails, $restrictions ) {
+		wc_doing_it_wrong(
+			'WC_Cart::is_coupon_emails_allowed',
+			'This method has been deprecated and will be removed soon. Use Automattic\WooCommerce\Utilities\DiscountsUtil::is_coupon_emails_allowed instead.',
```

Comment:
We don't seem to be consistent on this across the codebase, but I think it may be best to localize this message (if a merchant notices the warning, for instance, we probably want them to see it and read it in their own language).

```suggestion
			__( 'This method has been deprecated and will be removed soon. Use Automattic\WooCommerce\Utilities\DiscountsUtil::is_coupon_emails_allowed instead.', 'woocommerce' ),
```